Source/ProjectRimFactory/Common/HarmonyPatches/Patch_DSU_OrbitalTrade.cs

```diff
@@ -0,0 +1,108 @@
+        {
+            predicateClass = typeof(PassingShip).GetNestedTypes(HarmonyLib.AccessTools.all)
+               .FirstOrDefault(t => t.FullName.Contains("c__DisplayClass24_0"));
+            if (predicateClass == null)
+            {
+                Log.Error("PRF Harmony Error - predicateClass == null for Patch_PassingShip_DSUisTradebeacon.TargetMethod()");
+                return null;
+            }
+
+            var m = predicateClass.GetMethods(AccessTools.all)
+                                 .FirstOrDefault(t => t.Name.Contains("b__1"));
+            if (m == null)
+            {
+                Log.Error("PRF Harmony Error - m == null for Patch_PassingShip_DSUisTradebeacon.TargetMethod()");
+            }
+            return m;
+        }
+
+        static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+
+            Type hiddenClass = typeof(PassingShip).GetNestedTypes(HarmonyLib.AccessTools.all)
+               .FirstOrDefault(t => t.FullName.Contains("c__DisplayClass24_0"));
+
+            bool foundLocaterString = false;
+            foreach (var instruction in instructions)
+            {
+
+                //Patch shall change:
+                //if (!Building_OrbitalTradeBeacon.AllPowered(<>4__this.Map).Any())
+                //
+                //To:
+                //if (!Building_OrbitalTradeBeacon.AllPowered(<>4__this.Map).Any() && !Building_MassStorageUnitPowered.AllPowered(<>4__this.Map).Any() )
+
+
+                //Find the refrence Point
+                if (instruction.opcode == OpCodes.Call && (instruction.operand as MethodInfo) == AccessTools.Method(typeof(Building_OrbitalTradeBeacon), nameof(Building_OrbitalTradeBeacon.AllPowered)))
+                {
+                    foundLocaterString = true;
+                }
+
+                //Find the Check
+                if (instruction.opcode == OpCodes.Brtrue_S && foundLocaterString)
+                {
+                    //Keep the Inctruction
+                    yield return instruction;
+                    //this.Map
+                    yield return new CodeInstruction(OpCodes.Ldarg_0);
+                    yield return new CodeInstruction(OpCodes.Ldfld, AccessTools.Field(hiddenClass, "<>4__this"));
+                    yield return new CodeInstruction(OpCodes.Call, AccessTools.PropertyGetter(typeof(PassingShip), "Map"));
+                    //Call --> Building_MassStorageUnitPowered.AnyPowerd with the above as an argument
+                    yield return new CodeInstruction(OpCodes.Call, HarmonyLib.AccessTools
+                        .Method(typeof(Building_MassStorageUnitPowered) ,nameof(Building_MassStorageUnitPowered.AnyPowerd), new[] { typeof(Map)} ));
+                    yield return new CodeInstruction(OpCodes.Brtrue_S, instruction.operand);
+                    continue;
+
+                }
+                //Keep the other instructions
```

Comment:
`foundLocatorString` is still true - this means that if anything else uses `Brtrue_S` later in the code (or is added next year in some Ludeon update, or gets added by someone else's Harmony patch), this will get added again...possibly causing problems.

It's unlikely, but possible.  Setting `foundLocatorString` to false should take care of that.